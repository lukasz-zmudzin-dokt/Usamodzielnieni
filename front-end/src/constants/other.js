const DEFAULT_INPUT = "-- Wybierz --";

export { DEFAULT_INPUT };
