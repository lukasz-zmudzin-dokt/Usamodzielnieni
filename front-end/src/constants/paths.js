const paths = {
    DASHBOARD: "/",
    CV_CREATOR: "/cvEditor",
    CV_EDITOR: "/cvEditor/:id",
    REGISTER: "/newAccount/",
    REGISTER_ADMIN: "/newAccount/:role(staff)",
    FOOTER: "/footer",
    LOGIN: "/login",
    USER: "/user",
    OFFER_FORM: "/offerForm/:id?",
    JOB_OFFERS: "/jobOffers",
    CV_APPROVAL: "/cvApproval",
    MY_OFFERS: "/myOffers",
    CONTACT_PAGE: "/contact",
    BLOG_POST: "/blog/blogpost/:id",
    BLOG_PAGE: "/blog",
    BLOG_FORM: "/blog/newPost",
    BLOG_EDIT: "/blog/newPost/:id",
    JOB_OFFER_DETAILS: "/jobOffers/:id",
    MY_CVS: "/myCVs",
    USER_APPROVAL: "/userApproval",
    OFFER_APPROVAL: "/offerApproval"
};
export {paths};