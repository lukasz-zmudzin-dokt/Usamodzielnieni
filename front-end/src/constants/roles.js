import { staffTypes } from "./staffTypes";

export const adminGroup = [
  {
    name: staffTypes.VERIFICATION,
    placeholder: "Weryfikacja użytkowników",
  },
  {
    name: staffTypes.CV,
    placeholder: "Weryfikacja CV",
  },
  {
    name: staffTypes.JOBS,
    placeholder: "Weryfikacja ofert pracy",
  },
  {
    name: staffTypes.BLOG_CREATOR,
    placeholder: "Kreator postów na blogu",
  },
  {
    name: staffTypes.BLOG_MODERATOR,
    placeholder: "Moderator bloga / designer",
  },
  {
    name: staffTypes.GUEST,
    placeholder: "Konto gościa",
  },
  {
    name: staffTypes.CHAT,
    placeholder: "Dostęp do czatu",
  },
];

export const commonGroup = [
  {
    name: "Podopiecznym",
  },
  {
    name: "Pracodawcą",
  },
];
