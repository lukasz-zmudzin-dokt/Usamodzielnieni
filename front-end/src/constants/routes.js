import {
  UserProfilePage,
  LoginPage,
  RegisterPage,
  Footer,
  CVEditorPage,
  Menu,
  OfferForm,
  MyOffersPage,
  CVApprovalPage
} from "Views";

const paths = {
  DASHBOARD: "/",
  CV_EDITOR: "/cvEditor",
  REGISTER: "/newAccount",
  FOOTER: "/footer",
  LOGIN: "/login",
  USER: "/user",
  OFFER_FORM: "/offerForm",
  JOB_OFFERS: "/jobOffers",
  CV_APPROVAL: "/cvApproval",
  MY_OFFERS: "/myOffers"
};

export const userTypes = {
  STANDARD: "Standard",
  STAFF: "Staff",
  EMPLOYER: "Employer"
};

export default [
  {
    path: paths.DASHBOARD,
    component: Menu,
    exact: true
  },
  {
    path: paths.CV_EDITOR,
    component: CVEditorPage,
    isPrivate: true,
    type: userTypes.STANDARD
  },
  {
    path: paths.REGISTER,
    component: RegisterPage,
    exact: true
  },
  {
    path: paths.FOOTER,
    component: Footer,
    exact: true
  },
  {
    path: paths.LOGIN,
    component: LoginPage,
    exact: true
  },
  {
    path: paths.USER,
    component: UserProfilePage,
    isPrivate: true,
    type: undefined // jeżeli jest undefined to znaczy że jest dostępne dla wszystkich typów konta
  },
  {
    path: paths.JOB_OFFERS,
    component: UserProfilePage, // tu trzeba zmienić komponent
    isPrivate: true,
    type: userTypes.EMPLOYER
  },
  {
    path: paths.OFFER_FORM,
    component: OfferForm,
    isPrivate: true,
    type: userTypes.EMPLOYER
  },
  {
    path: paths.CV_APPROVAL,
    component: CVApprovalPage,
    isPrivate: true,
    type: userTypes.STAFF
  },
  {
    path: paths.MY_OFFERS,
    component: MyOffersPage,
    isPrivate: true,
    type: userTypes.EMPLOYER
  }
];

export { paths };
