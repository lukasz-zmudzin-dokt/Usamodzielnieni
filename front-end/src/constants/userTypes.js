export const userTypes = {
  STANDARD: "standard",
  STAFF: "staff",
  EMPLOYER: "employer",
};
