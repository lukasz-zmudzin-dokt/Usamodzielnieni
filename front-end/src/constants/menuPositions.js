import { userTypes } from "./userTypes";
import { staffTypes } from "./staffTypes";
import { paths } from "./paths";

export default [       
    {
        name: "Kreator CV",
        path: paths.CV_CREATOR,
        allowed: [userTypes.STANDARD],
        verified: true
    },
    {
        name: "Moje CV",
        path: paths.MY_CVS,
        allowed: [userTypes.STANDARD],
        verified: true
    },
    {
        name: "Akceptacja CV",
        path: paths.CV_APPROVAL,
        allowed: [staffTypes.CV],
        verified: true
    },
    {
        name: "Oferty pracy",
        path: paths.JOB_OFFERS,
        allowed: [userTypes.STANDARD, userTypes.STAFF, userTypes.EMPLOYER]
    },
    {
        name: "Dodaj ofertę",
        path: paths.OFFER_FORM,
        allowed: [userTypes.EMPLOYER],
        verified: true
    },
    {
        name: "Moje oferty",
        path: paths.MY_OFFERS,
        allowed: [userTypes.EMPLOYER],
        verified: true
    },
    {
        name: "Blogi",
        path: paths.BLOG_PAGE,
        allowed: [userTypes.STANDARD, userTypes.STAFF, userTypes.EMPLOYER]
    },
    {
        name: "Dodaj post",
        path: paths.BLOG_FORM,
        allowed: [staffTypes.BLOG_CREATOR],
        verified: true
    },
    {
        name: "Telefony",
        path: paths.CONTACT_PAGE,
        allowed: undefined, //wszyscy są upoważnieni - zalogowani i niezalogowani
    },
    {
        name: "Kroki",
        path: paths.STEPS,
        allowed: undefined //wszyscy są upoważnieni - zalogowani i niezalogowani
    },
    {
        name: "Chat",
        path: paths.CHATS,
        allowed: [userTypes.STANDARD, userTypes.STAFF, userTypes.EMPLOYER],
        verified: true,
    }
];
