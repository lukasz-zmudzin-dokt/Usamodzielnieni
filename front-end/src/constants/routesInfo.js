import { userTypes } from "./userTypes";
import { staffTypes } from "./staffTypes";
import { paths } from "./paths";

export default [
  {
    path: paths.DASHBOARD,
    exact: true,
  },
  {
    path: paths.CV_CREATOR,
    isPrivate: true,
    type: [userTypes.STANDARD, userTypes.STAFF],
    group: [staffTypes.GUEST, staffTypes.BLOG_MODERATOR],
    exact: true,
    userVerified: true,
  },
  {
    path: paths.CV_EDITOR,
    isPrivate: true,
    type: userTypes.STANDARD,
    exact: true,
    userVerified: true,
  },
  {
    path: paths.REGISTER,
    exact: true,
  },
  {
    path: paths.REGISTER_ADMIN,
    exact: true,
    isPrivate: true,
    type: userTypes.STAFF,
    group: staffTypes.VERIFICATION,
    userVerified: true,
  },
  {
    path: paths.FOOTER,
    exact: true,
  },
  {
    path: paths.LOGIN,
    exact: true,
  },
  {
    path: paths.USER,
    isPrivate: true,
    type: undefined, // jeżeli jest undefined to znaczy że jest dostępne dla wszystkich typów konta
  },
  {
    path: paths.JOB_OFFERS,
    exact: true,
    type: undefined,
  },
  {
    path: paths.JOB_OFFER_DETAILS,
    exact: true,
    type: undefined,
  },
  {
    path: paths.OFFER_FORM,
    isPrivate: true,
    type: [userTypes.EMPLOYER, userTypes.STAFF],
    group: [staffTypes.GUEST],
    userVerified: true,
  },
  {
    path: paths.CV_APPROVAL,
    isPrivate: true,
    type: userTypes.STAFF,
    group: staffTypes.CV,
    userVerified: true,
  },
  {
    path: paths.MY_OFFERS,
    isPrivate: true,
    type: userTypes.EMPLOYER,
    userVerified: true,
  },
  {
    path: paths.CONTACT_PAGE,
  },
  {
    path: paths.BLOG_POST,
    exact: true,
  },
  {
    path: paths.BLOG_PAGE,
    exact: true,
  },
  {
    path: paths.BLOG_FORM,
    isPrivate: true,
    type: userTypes.STAFF,
    group: [staffTypes.BLOG_CREATOR],
    exact: true,
    userVerified: true,
  },
  {
    path: paths.BLOG_EDIT,
    isPrivate: true,
    type: userTypes.STAFF,
    group: staffTypes.BLOG_CREATOR,
    exact: true,
    userVerified: true,
  },
  {
    path: paths.MY_CVS,
    isPrivate: true,
    type: userTypes.STANDARD,
    userVerified: true,
  },
  {
    path: paths.USER_APPROVAL,
    isPrivate: true,
    type: userTypes.STAFF,
    group: staffTypes.VERIFICATION,
    userVerified: true,
  },
  {
    path: paths.STEPS,
    isPrivate: true,
  },
  {
    path: paths.CV_CORRECTION,
    isPrivate: true,
    type: userTypes.STAFF,
    group: staffTypes.CV,
  },
  {
    path: paths.CHATS,
    isPrivate: true,
    type: undefined,
    group: staffTypes.CHAT,
    exact: true,
    userVerified: true,
  },
  {
    path: paths.CHAT_DETAILS,
    isPrivate: true,
    type: undefined,
    group: staffTypes.CHAT,
    exact: true,
    userVerified: true,
  },
  {
    path: paths.USER_LIST,
    isPrivate: true,
    type: userTypes.STAFF,
    exact: true,
    userVerified: true,
  },
  {
    path: paths.OFFER_APPROVAL,
    isPrivate: true,
    type: userTypes.STAFF,
    group: staffTypes.JOBS,
  },
  {
    path: paths.CHANGE_DATA,
    isPrivate: true,
    type: userTypes.STAFF,
    group: undefined,
  },
  {
    path: paths.PASSWORD_RESET,
    type: undefined,
    exact: true,
  },
  {
    path: paths.NEW_PASSWORD,
    type: undefined,
    exact: true,
  },
  {
    path: paths.BLOG_PAGE_FILTERED,
    type: undefined,
    exact: true
  }
];
