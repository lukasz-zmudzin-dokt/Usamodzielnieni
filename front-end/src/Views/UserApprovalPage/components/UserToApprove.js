import React, { useContext, useEffect, useState, useRef } from "react";
import { Alert, Button, Card, ListGroup, Row } from "react-bootstrap";
import { UserContext, AlertContext } from "context";
import {
  getUserDetails,
  setUserApproved,
  setUserRejected,
} from "Views/UserApprovalPage/functions/apiCalls";
import { DetailsItem, DeletionModal } from "components";
import { userTypes } from "constants/userTypes";

const UserToApprove = ({ user, activeUser }) => {
  const context = useContext(UserContext);
  const alertC = useRef(useContext(AlertContext));
  const [userDetails, setUserDetails] = useState([]);
  const [userDetailsFacilityAddress, setUserDetailsFacilityAddress] = useState(
    []
  );
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(false);

  const [showRejectModal, setShowReject] = useState(false);
  const [rejectConfirmed, setRejectConfirmed] = useState(false);
  //const [showAcceptModal, setShowAccept] = useState(false);
  //const [acceptConfirmed, setAcceptConfirmed] = useState(false);

  useEffect(() => {
    const loadUserDetails = async (token, userId) => {
      setLoading(true);
      try {
        let res = await getUserDetails(token, userId);
        setUserDetails(res);
        if (user.type === userTypes.STANDARD) {
          setUserDetailsFacilityAddress(res.facility_address);
        } else {
          setUserDetailsFacilityAddress(res.company_address);
        }
      } catch (err) {
        setError(true);
      }
      setLoading(false);
    };
    if (user.id === activeUser) {
      loadUserDetails(context.token, user.id);
    }
    if (rejectConfirmed) rejectUser(context.token, user.id);
  }, [context.token, user.id, user.type, activeUser, rejectConfirmed]);

  const approveUser = async (e, token, userId) => {
    e.preventDefault();
    try {
      let res = await setUserApproved(token, userId);
      if (res.message === "Użytkownik został pomyślnie zweryfikowany") {
        alertC.current.showAlert(res.message, "success");
      }
    } catch (err) {
      setError(true);
      alertC.current.showAlert("Nie udało się zweryfikować użytkownika");
    }
  };

  //const rejectUser = async (e, token, userId) => {
  const rejectUser = async (token, userId) => {
    let res;
    try {
      res = await setUserRejected(token, userId);
      console.log(res);
      if (res.message === "Użytkownik został pomyślnie odrzucony") {
        alertC.current.showAlert(res.message, "success");
      }
    } catch (err) {
      setError(true);
      alertC.current.showAlert("Nie udało się odrzucić użytkownika");
    }
    setRejectConfirmed(false);
  };

  const handleOnClick = (e) => {
    if (e.target.id === "reject") setShowReject(true);
    /*else if(e.target.id === "accept")
            setShowAccept(true);*/
  };

  const address =
    user.type === userTypes.STANDARD ? (
      <DetailsItem label="Adres">
        <p>{userDetails.facility_name}</p>
        <p>
          {userDetailsFacilityAddress.street}{" "}
          {userDetailsFacilityAddress.street_number}
        </p>
        <p>
          {userDetailsFacilityAddress.postal_code}{" "}
          {userDetailsFacilityAddress.city}
        </p>
      </DetailsItem>
    ) : (
      <>
        <DetailsItem label="Adres">
          <p>{userDetails.company_name}</p>
          <p>
            {userDetailsFacilityAddress.street}{" "}
            {userDetailsFacilityAddress.street_number}
          </p>
          <p>
            {userDetailsFacilityAddress.postal_code}{" "}
            {userDetailsFacilityAddress.city}
          </p>
        </DetailsItem>
        <DetailsItem label="NIP">{userDetails.nip}</DetailsItem>
      </>
    );

  const message = loading ? (
    <Alert className="mb-0" variant="info">
      Ładuję...
    </Alert>
  ) : error ? (
    <Alert className="mb-0" variant="danger">
      Nie udało się załadować danych użytkownika.
    </Alert>
  ) : null;

  /*if(acceptConfirmed)
        approveUser(context.token, user.id);*/

  if (message) {
    return message;
  } else {
    return (
      <Card.Body>
        <DeletionModal
          show={showRejectModal}
          setShow={setShowReject}
          delConfirmed={setRejectConfirmed}
          question={"Czy na pewno chcesz odrzucić tego użytkownika?"}
          confirmLabel={"Odrzuć"}
          cancelLabel={"Anuluj"}
        />
        <ListGroup variant="flush">
          <ListGroup.Item>
            <Row>
              <DetailsItem md={4} label="Nazwa użytkownika">
                {userDetails.username}
              </DetailsItem>
              <DetailsItem label="Imię">{userDetails.first_name}</DetailsItem>
              <DetailsItem label="Nazwisko">
                {userDetails.last_name}
              </DetailsItem>
            </Row>
          </ListGroup.Item>
          <ListGroup.Item>
            <Row>
              <DetailsItem label="Email">{userDetails.email}</DetailsItem>
              <DetailsItem label="Telefon">
                {userDetails.phone_number}
              </DetailsItem>
            </Row>
          </ListGroup.Item>
          <ListGroup.Item>
            <Row>{address}</Row>
          </ListGroup.Item>
          <ListGroup.Item>
            <Row className="justify-content-center">
              <Button
                onClick={(e) => approveUser(e, context.token, user.id)}
                variant="success"
                id="accept"
              >
                Akceptuj
              </Button>
              <Button
                onClick={(e) => handleOnClick(e)}
                variant="danger"
                className="ml-3"
                id="reject"
              >
                Odrzuć
              </Button>
            </Row>
          </ListGroup.Item>
        </ListGroup>
      </Card.Body>
    );
  }
};

export default UserToApprove;
