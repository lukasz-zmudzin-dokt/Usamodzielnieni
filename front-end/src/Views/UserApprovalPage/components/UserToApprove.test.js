import React from "react";
import { render, waitForElement, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import UserToApprove from "./UserToApprove";
import { userTypes } from "constants/userTypes";

describe("UserApproval", () => {
  let failFetch;
  let fetchUserType;
  let postType;

  let user = {
    standard: {
      email: "abc@gmail.com",
      id: "2949ad29-27da-49a0-aba2-1aa7b5bfa20b",
      type: userTypes.STANDARD,
      username: "standard0",
    },
    employer: {
      email: "string@aaa.aaa",
      id: "e8ac8431-cc60-423d-a044-9d048285f2ee",
      type: userTypes.EMPLOYER,
      username: "string",
    },
  };
  let apiUserDetails = {
    standard: {
      email: "abc@gmail.com",
      facility_address: {
        city: "Warszawa",
        postal_code: "11-123",
        street: "Aleja Niepodległości",
        street_number: "20",
      },
      facility_name: "facility",
      first_name: "Jan",
      id: "2949ad29-27da-49a0-aba2-1aa7b5bfa20b",
      last_name: "Kowalski",
      phone_number: "+48123123123",
      username: "standard0",
    },
    employer: {
      company_address: {
        city: "Ten test i tak nie przejdzie",
        postal_code: "69-123",
        street: "string",
        street_number: "693",
      },
      company_name: "string",
      email: "string@aaa.aaa",
      first_name: "string",
      id: "e8ac8431-cc60-423d-a044-9d048285f2ee",
      last_name: "string",
      nip: "5555555555",
      phone_number: "+48123456789",
      username: "string",
    },
  };

  beforeAll(() => {
    global.fetch = jest.fn().mockImplementation((input, init) => {
      return new Promise((resolve, reject) => {
        if (failFetch) {
          resolve({ status: 500 });
        }
        switch (init.method) {
          case "GET":
            switch (fetchUserType) {
              case userTypes.STANDARD:
                resolve({
                  status: 200,
                  json: () => Promise.resolve(apiUserDetails.standard),
                });
                break;
              case userTypes.EMPLOYER:
                resolve({
                  status: 200,
                  json: () => Promise.resolve(apiUserDetails.employer),
                });
                break;
              default:
                reject({});
                break;
            }
            break;
          case "POST":
            switch (postType) {
              case "Approve":
                resolve({
                  status: 200,
                  json: () => Promise.resolve("User successfully verified."),
                });
                break;
              case "Reject":
                resolve({
                  status: 200,
                  json: () =>
                    Promise.resolve(
                      "User status successfully set to not verified."
                    ),
                });
                break;
              default:
                reject({});
                break;
            }
            break;
          default:
            reject({});
            break;
        }
      });
    });
  });

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should match snapshot (Standard type user) ", async () => {
    failFetch = false;
    fetchUserType = userTypes.STANDARD;

    const { container, getByText } = render(
      <MemoryRouter>
        <UserToApprove user={user.standard} activeUser={user.standard.id} />
      </MemoryRouter>
    );
    await waitForElement(() => getByText("11-123 Warszawa"));
    expect(container).toMatchSnapshot();
  });

  it("should match snapshot (Employer type user) ", async () => {
    failFetch = false;
    fetchUserType = userTypes.EMPLOYER;

    const { container, getByText } = render(
      <MemoryRouter>
        <UserToApprove user={user.employer} activeUser={user.employer.id} />
      </MemoryRouter>
    );
    await waitForElement(() =>
      getByText("69-123 Ten test i tak nie przejdzie")
    );
    expect(container).toMatchSnapshot();
  });

  it("should view alert at api fail", async () => {
    failFetch = true;
    const { getByText } = render(
      <MemoryRouter>
        <UserToApprove user={user.standard} activeUser={user.standard.id} />
      </MemoryRouter>
    );

    await waitForElement(() => getByText("Ups, wystąpił błąd..."));
    expect(getByText("Ups, wystąpił błąd...")).toBeInTheDocument();
  });

  it("should accept user", async () => {
    failFetch = false;
    fetchUserType = userTypes.STANDARD;
    postType = "Approve";
    const { getByText } = render(
      <MemoryRouter>
        <UserToApprove user={user.standard} activeUser={user.standard.id} />
      </MemoryRouter>
    );
    await expect(fetch).toHaveBeenCalledWith(
      "https://usamo-back.herokuapp.com/account/admin/user_details/2949ad29-27da-49a0-aba2-1aa7b5bfa20b/",
      {
        headers: {
          Authorization: "token undefined",
          "Content-Type": "application/json",
        },
        method: "GET",
      }
    );
    await waitForElement(() => getByText("11-123 Warszawa"));
    fireEvent.click(getByText("Akceptuj"));
    await expect(fetch).toHaveBeenCalledWith(
      "https://usamo-back.herokuapp.com/account/admin/user_admission/2949ad29-27da-49a0-aba2-1aa7b5bfa20b/",
      {
        headers: {
          Authorization: "token undefined",
          "Content-Type": "application/json",
        },
        method: "POST",
      }
    );
    await waitForElement(() => getByText("Konto zatwierdzone pomyślnie."));
    expect(getByText("Konto zatwierdzone pomyślnie.")).toBeInTheDocument();
  });

  it("should reject user", async () => {
    failFetch = false;
    fetchUserType = userTypes.STANDARD;
    postType = "Reject";
    const { getByText } = render(
      <MemoryRouter>
        <UserToApprove user={user.standard} activeUser={user.standard.id} />
      </MemoryRouter>
    );
    await expect(fetch).toHaveBeenCalledWith(
      "https://usamo-back.herokuapp.com/account/admin/user_details/2949ad29-27da-49a0-aba2-1aa7b5bfa20b/",
      {
        headers: {
          Authorization: "token undefined",
          "Content-Type": "application/json",
        },
        method: "GET",
      }
    );
    await waitForElement(() => getByText("11-123 Warszawa"));
    fireEvent.click(getByText("Odrzuć"));
    await expect(fetch).toHaveBeenCalledWith(
      "https://usamo-back.herokuapp.com/account/admin/user_rejection/2949ad29-27da-49a0-aba2-1aa7b5bfa20b/",
      {
        headers: {
          Authorization: "token undefined",
          "Content-Type": "application/json",
        },
        method: "POST",
      }
    );
    await waitForElement(() => getByText("Konto odrzucone pomyślnie."));
    expect(getByText("Konto odrzucone pomyślnie.")).toBeInTheDocument();
  });
});
