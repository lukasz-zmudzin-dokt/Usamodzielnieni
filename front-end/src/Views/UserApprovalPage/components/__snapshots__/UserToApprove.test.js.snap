// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserApproval should match snapshot (Employer type user)  1`] = `
<div>
  <div
    class="card-body"
  >
    <div
      class="list-group list-group-flush"
    >
      <div
        class="list-group-item"
      >
        <div
          class="row"
        >
          <div
            class="detailsItem col-md-4"
          >
            <div
              class="detailsItem__label"
            >
              Nazwa użytkownika
            </div>
            <div
              class="detailsItem__value"
            >
              string
            </div>
          </div>
          <div
            class="detailsItem col"
          >
            <div
              class="detailsItem__label"
            >
              Imię
            </div>
            <div
              class="detailsItem__value"
            >
              string
            </div>
          </div>
          <div
            class="detailsItem col"
          >
            <div
              class="detailsItem__label"
            >
              Nazwisko
            </div>
            <div
              class="detailsItem__value"
            >
              string
            </div>
          </div>
        </div>
      </div>
      <div
        class="list-group-item"
      >
        <div
          class="row"
        >
          <div
            class="detailsItem col"
          >
            <div
              class="detailsItem__label"
            >
              Email
            </div>
            <div
              class="detailsItem__value"
            >
              string@aaa.aaa
            </div>
          </div>
          <div
            class="detailsItem col"
          >
            <div
              class="detailsItem__label"
            >
              Telefon
            </div>
            <div
              class="detailsItem__value"
            >
              +48123456789
            </div>
          </div>
        </div>
      </div>
      <div
        class="list-group-item"
      >
        <div
          class="row"
        >
          <div
            class="detailsItem col"
          >
            <div
              class="detailsItem__label"
            >
              Adres
            </div>
            <div
              class="detailsItem__value"
            >
              <p>
                string
              </p>
              <p>
                string
                 
                693
              </p>
              <p>
                69-123
                 
                Ten test i tak nie przejdzie
              </p>
            </div>
          </div>
          <div
            class="detailsItem col"
          >
            <div
              class="detailsItem__label"
            >
              NIP
            </div>
            <div
              class="detailsItem__value"
            >
              5555555555
            </div>
          </div>
        </div>
      </div>
      <div
        class="list-group-item"
      >
        <div
          class="justify-content-center row"
        >
          <button
            class="btn btn-success"
            type="button"
          >
            Akceptuj
          </button>
          <button
            class="ml-3 btn btn-danger"
            type="button"
          >
            Odrzuć
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`UserApproval should match snapshot (Standard type user)  1`] = `
<div>
  <div
    class="card-body"
  >
    <div
      class="list-group list-group-flush"
    >
      <div
        class="list-group-item"
      >
        <div
          class="row"
        >
          <div
            class="detailsItem col-md-4"
          >
            <div
              class="detailsItem__label"
            >
              Nazwa użytkownika
            </div>
            <div
              class="detailsItem__value"
            >
              standard0
            </div>
          </div>
          <div
            class="detailsItem col"
          >
            <div
              class="detailsItem__label"
            >
              Imię
            </div>
            <div
              class="detailsItem__value"
            >
              Jan
            </div>
          </div>
          <div
            class="detailsItem col"
          >
            <div
              class="detailsItem__label"
            >
              Nazwisko
            </div>
            <div
              class="detailsItem__value"
            >
              Kowalski
            </div>
          </div>
        </div>
      </div>
      <div
        class="list-group-item"
      >
        <div
          class="row"
        >
          <div
            class="detailsItem col"
          >
            <div
              class="detailsItem__label"
            >
              Email
            </div>
            <div
              class="detailsItem__value"
            >
              abc@gmail.com
            </div>
          </div>
          <div
            class="detailsItem col"
          >
            <div
              class="detailsItem__label"
            >
              Telefon
            </div>
            <div
              class="detailsItem__value"
            >
              +48123123123
            </div>
          </div>
        </div>
      </div>
      <div
        class="list-group-item"
      >
        <div
          class="row"
        >
          <div
            class="detailsItem col"
          >
            <div
              class="detailsItem__label"
            >
              Adres
            </div>
            <div
              class="detailsItem__value"
            >
              <p>
                facility
              </p>
              <p>
                Aleja Niepodległości
                 
                20
              </p>
              <p>
                11-123
                 
                Warszawa
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="list-group-item"
      >
        <div
          class="justify-content-center row"
        >
          <button
            class="btn btn-success"
            type="button"
          >
            Akceptuj
          </button>
          <button
            class="ml-3 btn btn-danger"
            type="button"
          >
            Odrzuć
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
